### PR TITLE
feat: starter building blocks for all users

### LINKED ISSUE #30903: [Task]: Make starter building blocks available to all users on all pages.

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
**Goal**
The starter building block on canvas experiment was a success, this PR removes the feature from behind a feature flag and makes it available to all users on the platform.

Summary of changes
- Remove feature flag condition in DropTarget component
- Clean up unnecessary state used to find newUsers and firstTimeUsers
- Remove redundant storage functions used for storing first time user and application id
- Remove showStarterTemplatesInsteadofBlankCanvas prop in signUpHelper and signUpSuccess
- 
#### PR fixes following issue(s)
Fixes #30903 

#### Type of change
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Simplified signup process by removing the option to choose starter templates or a blank canvas.
	- Streamlined onboarding component logic and storage management for a smoother user experience.
	- Optimized import organization and removed unused functions in the DropTargetComponent.
	- Enhanced test coverage for building blocks and starter templates in client-side templates.

### CODE REVIEW

File: app/client/src/layoutSystems/common/dropTarget/DropTargetComponent.tsx [hunk 1]
```diff
@@ -76,40 +68,14 @@ const StyledDropTarget = styled.div`
...
-  const currentApplicationId = useSelector(
-    (state: AppState) => state.ui.applications.currentApplication?.id,
-  );
-
   const shouldShowStarterTemplates = useMemo(
-    () =>
-      showStarterTemplatesInsteadofBlankCanvas &&
-      !isMobileCanvas &&
-      isUsersFirstApp &&
-      !isAirgappedInstance,
```
Comment (rahulbarwal) [inline app/client/src/layoutSystems/common/dropTarget/DropTargetComponent.tsx:97]: Lets not remove the feature flag right now. We still want to be able to remove this if required.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Over all looks good, Just small nit: lets preserve feature flag for now. We still want to be able to shut down, if we see any problems.

File: app/client/src/layoutSystems/common/dropTarget/DropTargetComponent.tsx (hunk 1 above)
Comment (jacquesikot) [reply app/client/src/layoutSystems/common/dropTarget/DropTargetComponent.tsx:97]: Flag has been added back

**Review by rahulbarwal — APPROVED** [review]

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress.

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**